metricshub-agent/src/main/java/org/sentrysoftware/metricshub/agent/helper/ConnectorVariablesLibraryParser.java

```diff
@@ -0,0 +1,288 @@
+			this.additionalConnectorConfig = additionalConnectorConfig;
+		}
+
+		@Override
+		public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+			// Skip this path if it is a directory or not a YAML file
+			if (Files.isDirectory(path) || !isYamlFile(path.toFile().getName())) {
+				return FileVisitResult.CONTINUE;
+			}
+			final ObjectMapper yamlMapper = new YAMLMapper();
+			final JsonNode connectorNode = yamlMapper.readTree(path.toFile());
+			if (!isConnector(connectorNode)) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Get the connector's file name
+			final String filename = path.getFileName().toString();
+			final String connectorId = filename.substring(0, filename.lastIndexOf('.'));
+
+			if (!connectorNode.toString().contains("${var::")) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Normalize additionalConnectors
+			normalizeAdditionalConnectors();
+
+			// Removing all the configurations that are not using this connector.
+			final Map<String, AdditionalConnector> filteredConnectors = additionalConnectorConfig
+				.entrySet()
+				.stream()
+				.filter(entry -> connectorId.equalsIgnoreCase(entry.getValue().getUses()))
+				.collect(
+					Collectors.toMap(
+						Map.Entry::getKey, // Keep the original key
+						Map.Entry::getValue // Keep the original AdditionalConnectorConfig as the value
+					)
+				);
+
+			// Construct a variables map from the default connector variables.
+			final Map<String, String> defaultVariables = new HashMap<>(getDefaultConnectorVariables(connectorNode));
+
+			// Parse the connector even if it is not configured as an additional connector.
+			// This ensures that the connector will function with the default variables if the user forces it.
+			if (filteredConnectors.isEmpty()) {
+				parseConnectorWithModifier(path, defaultVariables, connectorId, filename, connector -> {});
+			}
+
+			// For each configuration, we create a new custom connector and a new variables map to be used in the connector update.
+			for (final Entry<String, AdditionalConnector> connectorConfigurationEntry : filteredConnectors.entrySet()) {
+				final String additionalConnectorId = connectorConfigurationEntry.getKey();
+				final AdditionalConnector additionalConnectorValue = connectorConfigurationEntry.getValue();
+
+				// Add the connector to the host connectors set.
+				connectorsParsingResult
+					.getHostConnectors()
+					.add(additionalConnectorValue.isForce() ? "+" + additionalConnectorId : additionalConnectorId);
+
+				// Retrieve and use default connector variables on this connector for this configuration.
+				final Map<String, String> connectorVariables = new HashMap<>(defaultVariables);
+
+				// Override the default connector variables by the connector variables that the user configured.
+				final Map<String, String> configuredVariables = additionalConnectorValue.getVariables();
+				if (configuredVariables != null) {
```

Comment:
Should we set the default variables if there is no configured variables? and probably skipping the connector if there no configured/default variables at all.

Reply:
It would be logical to ignore these connectors since they won't return any relevant information.Their processing is time and CPU consuming 👎